pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -287,8 +292,81 @@ func (r *ReconcileCnsFileAccessConfig) Reconcile(request reconcile.Request) (rec
 			return reconcile.Result{RequeueAfter: timeout}, nil
 		}
 
+		cnsFileVolumeClientInstance, err := cnsfilevolumeclient.GetFileVolumeClientInstance(ctx)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get CNSFileVolumeClient instance. Error: %+v", err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		err = cnsFileVolumeClientInstance.AddClientVMToIPList(ctx, instance.Namespace+"/"+instance.Spec.PvcName, instance.Spec.VMName, vm.Status.VmIp)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to add VM %q with IP %q to IPList. Error: %+v", vm.Name, vm.Status.VmIp, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		log.Debugf("Successfully added VM IP %q to IPList for CnsFileAccessConfig request with name: %q on namespace: %q", vm.Status.VmIp, instance.Name, instance.Namespace)
+
+		cnsVolumeID := cnstypes.CnsVolumeId{
+			Id: volumeID,
+		}
+		vSanFileShareNetPermissions := make([]vsanfstypes.VsanFileShareNetPermission, 0)
+
+		cfg, err := config.GetConfig()
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get config. Err: %+v", err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		// create a new dynamic client for config
+		dynamicClient, err := dynamic.NewForConfig(cfg)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to create client using config. Err: %+v", err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		// TODO:
+		// Identify the network provider (NSX-T or VDS). Assuming NSX-T configuration for now
+		tkgVMIP, err := cnsoperatorutil.GetTKGVMIP(ctx, r.vmOperatorClient, dynamicClient, vm.Namespace, vm.Name, true)
+		if err != nil {
+			msg := fmt.Sprintf("Failed to get external facing IP address for VM %q/%q. Err: %+v", vm.Namespace, vm.Name, err)
+			log.Error(msg)
+			setInstanceError(ctx, r, instance, msg)
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+		log.Debugf("Found tkg VMIP %q for VM %q in namespace %q", tkgVMIP, vm.Name, vm.Namespace)
+
 		// TODO:
-		// Get the network configuration details and configuring File share ACLs
+		// Fetch net permissions from config
+		vsanFileShareAccessType := vsanfstypes.VsanFileShareAccessTypeREAD_WRITE
```

Comment:
This step only occurs if `tkgVMIP` has not already been configured for this particular file volume

Reply:
Moving the configure ACL logic to AddClientVMToIPList() which has the same check.